File: 2405.1/jf_syn/basic/WeakEncryptionInsufficientKeySizeChecker/WeakEncryptionInsufficientKeySize.java
package basic;

import java.security.KeyPair;
import java.security.KeyPairGenerator;

public class WeakEncryptionInsufficientKeySize {
    int keySize = 512;

    public void unsafe() throws Exception {
        int ks = 511;
        int ks2 = 11;

        KeyPairGenerator keyPairGenerator1 = KeyPairGenerator.getInstance("AES");
        keyPairGenerator1.initialize(ks2 + 2, null); // @violation

        KeyPairGenerator keyPairGenerator2 = KeyPairGenerator.getInstance("SEED");
        keyPairGenerator2.initialize(64, null); // @violation

        KeyPairGenerator keyPairGenerator3 = KeyPairGenerator.getInstance("RSA");
        keyPairGenerator3.initialize(512, null); // @violation
        keyPairGenerator3.initialize(ks + keySize); // @violation
        keyPairGenerator3.initialize(keySize); // @violation

        KeyPairGenerator keyPairGenerator4 = KeyPairGenerator.getInstance("EC");
        keyPairGenerator4.initialize(ks2, null); // @violation
    }

    public void safe() throws Exception {
        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("SHA-1");
        keyPairGenerator.initialize(128, null);
        KeyPair keyPair = keyPairGenerator.generateKeyPair();
    }
}

File: 2405.1/kotlinchk/security/WeakEncryptionInsufficientKeySizeChecker/WeakEncryptionInsufficientKeySize.kt
package testcase.rule.security.WeakEncryption.InsufficientKeySizeChecker

import java.security.KeyPairGenerator

object TestCase {
	fun violation1() : String {
		val keyGen : KeyPairGenerator
		if(true) {
			keyGen = KeyPairGenerator.getInstance("RSA")
			keyGen.initialize(512)					// @violation
		}
		else {
			keyGen = KeyPairGenerator.getInstance("SHA-256")
			keyGen.initialize(512)					// good
		}

		val key = keyGen.generateKeyPair()
		return key.public.toString()
	}

	fun violation2() : String {
		val keyGen = KeyPairGenerator.getInstance("RSA")
		keyGen.initialize(1024 + 1023)				// @violation
		val key = keyGen.generateKeyPair()
		return key.public.toString()
	}

	fun violation3() : String {
		val keyGen = KeyPairGenerator.getInstance("RSA")
		val keySize = 512 + 1024
		keyGen.initialize(keySize)					// @violation
		val key = keyGen.generateKeyPair()
		return key.public.toString()
	}

	fun good1() : String {
		val keyGen = KeyPairGenerator.getInstance("RSA")
		keyGen.initialize(2048)						// good
		val key = keyGen.generateKeyPair()
		return key.public.toString()
	}

	fun good2() : String {
		val keyGen = KeyPairGenerator.getInstance("RSA")
		val keySize = 1023 * 2 + 100
		keyGen.initialize(keySize)					// good
		val key = keyGen.generateKeyPair()
		return key.public.toString()
	}
}